### PR TITLE
Wrap S3 multipart upload exception

### PR DESCRIPTION
**Pull request by artonge**
The returned HTTP code is now [502](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502) instead of [500](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500) in case of timeout during S3 bucket upload.

Hopefully, this would help unofficial clients to deal with this kind of error in a better way.

### CODE REVIEW

File: apps/dav/lib/Connector/Sabre/Exception/BadGateway.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @copyright Copyright (c) 2016, ownCloud, Inc.
```
Comment (PVince81) [inline apps/dav/lib/Connector/Sabre/Exception/BadGateway.php:3]: copy-pasta :ramen:

**Review by PVince81 — APPROVED** [review]
:+1: looks fine

I'm not 100% sure about "502 bad gateway" even though technically Nextcloud is like a proxy here. Let's see what the second reviewer thinks

File: apps/dav/lib/Connector/Sabre/Exception/BadGateway.php (hunk 1 above)
Comment (artonge) [reply apps/dav/lib/Connector/Sabre/Exception/BadGateway.php:3]: :see_no_evil:

**Comment by artonge** [comment]
> I'm not 100% sure about "502 bad gateway" even though technically Nextcloud is like a proxy here. Let's see what the second reviewer thinks

I would prefer the timeout code (408) but I didn't find a way to check if the error was from a timeout or something else.

**Review by ChristophWurst — APPROVED** [review]
:+1: :elephant:

**Comment by artonge** [comment]
CI failure unrelated

**Pull request merged**